nservicebus/upgrades/5to6.md

```diff
@@ -19,56 +19,20 @@ The minimum .NET version for NServiceBus Version 6 is .NET 4.5.2.
-
-snippet: 5to6-endpointNameRequired
-
-If the Version 5, derived endpoint name behavior, is required then use the following helper class.
-
-snippet: 5to6EndpointNameHelper
-
-
-## IStartableBus and Bus static class are obsolete
-
-In previous versions of NServiceBus to start a new instance of an endpoint the `Bus` static class and the `IStartableBus` interface. In Version 6 these two concepts have been replaced. More information can be found in the [hosting](/nservicebus/hosting/) section of the documentation.
-
-snippet:5to6-endpoint-start
-
-
-## IBus is obsolete
+## IBus, IStartableBus and the Bus Static class are now obsolete
 
 In previous versions of NServiceBus, to send or publish messages within a message handler or other extension interfaces, the message session (`IBus` interface in Versions 5 and below) was accessed via container injection. In Versions 6 injecting the message session is no longer required. Message handlers and other extension interfaces now provide context parameters such as `IMessageHandlerContext` or `IEndpointInstance` which give access to the same functions that used to be available via the `IBus` interface.
 
-
-## UnicastBus made internal
+For more details on the various scenarios when using IBus, see: [Migrating from IBus](/nservicebus/upgrades/5to6-moving-away-from-ibus.md).
 
 
-### Accessing the builder
-
-When using the `IBuilder` interface outside the infrastructure of NServiceBus it was possible to use a hack by casting the `IBus` interface to `UnicastBus` and then accessing the `Builder` property like this:
-
-    var builder = ((UnicastBus)bus).Builder
-
-This is no longer supported. It is advised to, instead of using `IBuilder` directly, use dependency injection via the container of choice.
-
-
-### Accessing the current message context
-
-Previously it was possible to access message parameters such `MessageId`, `ReplyToAddress` and the message headers via the `CurrentMessageContext` property on the `IBus` interface. These message properties are now available directly via the message handler context parameter.
-
-snippet: 5to6-messagecontext
-
-
-### Setting the host information
-
-Control over `HostInformation` was previously done using `UnicastBus.HostInformation`. This is now done using the more explicit API to set the host identifier, see `busConfiguration.UniquelyIdentifyRunningInstance()`.
+## [Endpoint](/nservicebus/endpoints/) Name is mandatory
 
+In Versions 6 and above endpoint name is mandatory.
 
-### Accessing ReadOnlySettings
+snippet: 5to6-endpointNameRequired
 
-`ReadOnlySettings` have been exposed on `UnicastBus.Settings`. The settings should only be accessed inside features, the pipeline and the start/stop infrastructure. Therefore accessing the settings over the `UnicastBus` is no longer supported.
+Use a helper class described in [Derived endpoint name behavior](/nservicebus/upgrades/5to6-endpoint-name-helper.md) to get the behavior that was previously available in Version 5.
```

Comment:
> Use a helper class ...

Is too imperative for my tastes. Is it possible that the reader isn't even aware of why they'd want the old behavior? Would it be worth having something like:

> The endpoint name is used as a logical identifier when sending or receiving messages. It is also used for determining the name of the input queue the endpoint will be bound to. See [Derived endpoint name]() for the algorithm used in Version 5 and below to select endpoint name if backwards compatibility is a concern.